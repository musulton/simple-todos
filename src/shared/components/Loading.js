import {ActivityIndicator, StatusBar, StyleSheet, View} from "react-native";
import React from "react";
import {useSelector} from "react-redux";

const Loading = () => {
    const isLoading = useSelector((state) => state.AppReducer.isLoading);
    return isLoading ? (
        <View style={styles.loading}>
            <StatusBar backgroundColor={'#00000080'}/>
            <ActivityIndicator size={"large"} color="#0000ff"/>
        </View>
    ) : (<View/>)
}
const styles = StyleSheet.create({
    loading: {
        flex: 1,
        position: 'absolute',
        left: 0,
        right: 0,
        top: 0,
        bottom: 0,
        opacity: 0.5,
        backgroundColor: 'black',
        justifyContent: 'center',
        alignItems: 'center'
    }
});
export default Loading;
