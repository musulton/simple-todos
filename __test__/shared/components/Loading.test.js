import {render, screen} from "@testing-library/react-native";

import Loading from "../../../src/shared/components/Loading";
import {useSelector} from "react-redux";

jest.mock("react-redux", () => ({
    useSelector: jest.fn(() => true)
}))

describe("Loading", () => {
    test("should render correctly when loading", () => {
        render(<Loading />)

        expect(screen.toJSON()).toMatchSnapshot()
    })

    test("should render correctly when is not loading", () => {
        useSelector.mockImplementation(() => false)

        render(<Loading />)

        expect(screen.toJSON()).toMatchSnapshot()
    })
})
