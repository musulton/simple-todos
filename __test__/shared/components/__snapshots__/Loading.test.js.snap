// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading should render correctly when is not loading 1`] = `<View />`;

exports[`Loading should render correctly when loading 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "black",
      "bottom": 0,
      "flex": 1,
      "justifyContent": "center",
      "left": 0,
      "opacity": 0.5,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <ActivityIndicator
    color="#0000ff"
    size="large"
  />
</View>
`;
